### PR TITLE
[diffusion] move SLA to attention_backend

### PR DESCRIPTION
**Pull request by IPostYellow**
## Motivation
move SLA to attention_backend

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by IPostYellow** [comment]
@mickqian @yhyang201 could you please add a CI tag？thanks.

**Review by mickqian — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/layers/attention/turbo_layer.py [hunk 1]
```diff
@@ -336,8 +221,26 @@ def set_context_parallel_group(self, group, stream):
...
+        self,
+        num_heads: int,
+        head_size: int,
+        attention_type: str,
+        topk: float,
+        supported_attention_backends: set[AttentionBackendEnum] | None = None,
+    ):
+        dtype = get_compute_dtype()
+        if attention_type == "sla":
+            get_global_server_args().attention_backend = "SLA_ATTN"
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/layers/attention/turbo_layer.py]: is there a better way to handle this?

**Review by mickqian — COMMENTED** [review]
please resolve gemini's issues

**Review by yhyang201 — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+        k_sum = torch.sum(k, dim=-2, keepdim=True)
+        return torch.matmul(q, kv) / (1e-5 + torch.matmul(q, k_sum.transpose(-1, -2)))
+
+    def forward(
+        self,
+        query: torch.Tensor,
+        key: torch.Tensor,
+        value: torch.Tensor,
+        attn_metadata: SparseLinearAttentionMetadata = None,
+    ) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py:149]: Not used; only for maintaining the forward parameter of  `AttentionImpl`

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+        # Combine sparse and linear attention
+        output = (o_s + o_l).to(dtype).transpose(1, 2)
+
+        return output
+
+
+class _attention(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, q, k, v, k_block_id, lut, topk, BLOCK_M, BLOCK_N, qk_scale=None):
```
[the remark this answers is not in the document]
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py:200]: Perhaps it is necessary to keep the names of both methods consistent to make it easier to understand what the methods actually do.

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        assert q.is_contiguous() and k.is_contiguous() and v.is_contiguous()
+        assert k_block_id.is_contiguous() and lut.is_contiguous()
```
[the remark this answers is not in the document]
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py:202]: the same as last

File: python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+            L,
+            M_BLOCKS,
+            D,
+            BLOCK_M,
+            BLOCK_N,
+            num_warps=4 if q.shape[-1] == 64 else 8,
+            num_stages=3,
+        )
+
+        ctx.save_for_backward(q, k, v, k_block_id, lut, lse, o_s)
```
[the remark this answers is not in the document]
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/backends/sparse_linear_attn.py:236]: the same as last

**Review by mickqian — CHANGES_REQUESTED** [review]
as above

File: python/sglang/multimodal_gen/runtime/layers/attention/turbo_layer.py (hunk 1 above)
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/turbo_layer.py]: OK, implemented in a better way.

[the remark this answers is not in the document]
Comment (IPostYellow) [reply python/sglang/multimodal_gen/runtime/layers/attention/turbo_layer.py]: fix

**Review by mickqian — COMMENTED** [review]
do we have a CI dedicated for this backend?

**Comment by IPostYellow** [comment]
> do we have a CI dedicated for this backend?

Yes, both `turbo_wan2_2_i2v_a14b_2gpu` and `turbo_wan2_1_t2v_1.3b` in cuda use this backend.

**Review by mickqian — APPROVED** [review]
